ocrd/ocrd/cli/workspace.py

```diff
@@ -38,7 +42,7 @@ def __init__(self, directory, mets_url, automatic_backup):
         self.directory = directory
         self.resolver = Resolver()
         self.mets_url = mets_url
-        self.mets_basename = basename(mets_url)
+        self.mets_basename = mets_basename if mets_basename else basename(mets_url)
```

Comment:
IMO this is incorrect: If the user called with `-d somewhere -M METS.XML`, then this will (issue the deprecation warning and) set `self.mets_url = somewhere/mets.xml` (instead of `somewhere/METS.XML`) and then set `self.mets_basename` correctly (but without effect).

I suggest to let `mets_basename` default to `mets.xml` and use that in the two conditions joining `mets_url`. You should also drop `self.mets_basename` from the context class to avoid mistakenly using it henceforth.

Reply:
e9e76bc I hope I didn't mess up the mets_basename logic for the Workspace constructor. Tested various combinations of `-m`/`-M`/`-d` on the command line and added tests to cover all conditionals for WorkspaceCtx.